localfs/emptydir_test.go

```diff
@@ -0,0 +1,118 @@
+package localfs
+
+import (
+	"os"
+	"path"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestDirectoryCleanup(t *testing.T) {
+	testDir, err := os.MkdirTemp("/tmp", "dirtest")
+	require.NoError(t, err)
+	defer os.RemoveAll(testDir)
+
+	makeDummyFile := func(filePath string) string {
+		fullPath := path.Join(testDir, filePath)
+		dir := path.Dir(fullPath)
+		require.NotEmpty(t, dir)
+		err := os.MkdirAll(dir, 0755)
+		require.NoError(t, err)
+		err = os.WriteFile(fullPath, []byte("don't delete this folder"), 0755)
+		require.NoError(t, err)
```

Comment:
Since you're using subtests below I believe you need to accept `t testing.TB` as an argument, otherwise you'll reference the wrong `*testing.T`.